### PR TITLE
Building heatpump dr integration

### PR DESCRIPTION
**Pull request by nick-harder**
this branch introduces the functionality of DSM units and their operation

### CODE REVIEW

File: assume/optitest.py
Comment (maurerle) [inline assume/optitest.py:1]: this should be moved to the test directory and converted to a pytest test

File: assume/common/outputs.py
Comment (maurerle) [inline assume/common/outputs.py:1]: Thats good :)

File: assume/common/base.py
Comment (maurerle) [inline assume/common/base.py:1]: remove Union here
maybe have technology always as a list[str]?

File: assume/common/forecasts.py [hunk 1]
```diff
@@ -127,6 +148,30 @@ def calc_forecast_if_needed(self):
...
+
+    def get_eom_price_forcast(self) -> pd.Series:
+        """
+        Returns the heating demand of the building
+
+        :rtype: pd.Series
+        """
+        return self.forecasts.get("price_EOM", pd.Series(0, self.index))
+
     def get_registered_market_participants(self, market_id):
```
Comment (maurerle) [inline assume/common/forecasts.py:175]: this whole section seems redundant.. Doesnt the same happen if i run self["price_EOM"] for example? Why do I need the getter for this?

File: assume/world.py [hunk 2]
```diff
@@ -223,7 +224,7 @@ async def async_add_unit(
         Create and add a new unit to the world.
 
         Params
-        ------
+        ------sssssssssssssssssssssssssssssssssssssssssss
```
Comment (maurerle) [inline assume/world.py:227]: remove

File: assume/world.py [hunk 3]
```diff
@@ -396,7 +397,7 @@ def add_unit(
                 id=id,
                 unit_type=unit_type,
                 unit_operator_id=unit_operator_id,
-                unit_params=unit_params,
+                unit_params=unit_params if unit_params is not None else {},
```
Comment (maurerle) [inline assume/world.py:400]: This should not be necessary, by having a better default for unit_params

**Review by maurerle — CHANGES_REQUESTED** [review]
I did not yet revise the new files, but looked at the changes to the existing framework

File: assume/units/dst_components.py [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+        @self.component_block.Constraint(time_steps)
+        def cooling_factor_constraint(m, t):
+            return m.power_in[t] == m.cool_out[t] / m.cooling_factor
+
+    def define_objective(self):
+        # Define the objective function specific to HeatPump if needed
+        pass
+
+
+class Storage:
```
Comment (maurerle) [inline assume/units/dst_components.py:152]: This should not be called Storage, as we already have a Storage class

**Comment by nick-harder** [comment]
stale. Reopen when active and updated

**Pull request closed without merging**